### DISCUSSION TITLE
Using AI ID instead of discord users id

### DISCUSSION THREAD

**Discussion by TFWIsaac** [JakyeRU/Larascord#61] [Ideas]
Hi,

i would like to make a suggestion:

Use the AI id instead of the discord users id and store the discord users id in a ceratin field.  You may ask now why?
Its maybe simple. I am using the laravel-debug-bar, so i can see every query wich was performed. So i have noticed that with a quite longer Discord-users-id a query take significant longer as a query with a "normal ID" (like 1, 253, 14521).

So in my case, my discord id is up to 19 digits. I assume that this can cause some perfomance problems in the future...

In the next days, i will setup two new fresh installations. Then i can show some examples.

**Comment by ValentinNeaga** [comment]
Hello @TFWIsaac,

Thanks for your suggestion. Seems like the performance issues are not caused by snowflakes, but because I set the `id` cast to a string in the user's model instead of an integer. (https://github.com/JakyeRU/Larascord/blob/main/src/Models/User.php#L48)

Can you make this modification and confirm that you're having the same performance issues?

**Reply by ValentinNeaga** [reply]
I have fixed it in #63 . Though it won't be live until the next release.
